### PR TITLE
PJS002 - Correct elements related to project (IVS-5)

### PR DESCRIPTION
**Pull request by Ghesselink**
https://github.com/buildingSMART/ifc-gherkin-rules/issues/111

Checking correct elements in both the `IfcRelDeclares` relationship as well as `IfcRelAggregates`

> NOTE The actual object occurrences used within the context of a project, such as walls, beams, air outlets, are assigned to a spatial hierarchy that is linked to the project using the aggregation hierarchy. See concept Spatial Decomposition for linking a spatial structure to the project.
>
> https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Project_Context/Project_Declaration/content.html

☝

(source: https://github.com/IfcOpenShell/IfcOpenShell/commit/2557580500042b115d97b422928abca9e3b116cd#commitcomment-132712129)

### CODE REVIEW

**Comment by evandroAlfieri** [comment]
IVS-5

File: features/Functional-parts.md [hunk 1]
```diff
@@ -25,4 +25,6 @@
 | SYS  |                                  | @SYS        |                                                  |
 | IFC  | IFC                              | @IFC        | Validating correct IFC Schema                    |
 | PSE  | Property Sets                    | @PSE        |                                                  |
-| APA  |                                  | @PSE        |                                                  |
\ No newline at end of file
+| APA  |                                  | @APA        |                                                  |
```
Comment (evandroAlfieri) [inline features/Functional-parts.md:28]: APA?

File: features/PJS002_Project-declaration.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@implementer-agreement
+@PJS
+@BLT
+@version1
+@E00010
+Feature: PJS002 - Project Declaration
```
Comment (evandroAlfieri) [inline features/PJS002_Project-declaration.feature:6]:
```suggestion
Feature: PJS002 - Correct elements related to project
```
Please adjust the file name accordingly

File: features/PJS002_Project-declaration.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  The rule verifies that the actual object occurrences (e.g. beams, walls) used within the context of a project
+  are not declared to to the project using the aggregation or declaration hierarchy.
+  See concept Spatial Decomposition for linking a spatial structure to the project.
+
+Scenario Outline: Validating correct elements declaration from IfcProject
```
Comment (evandroAlfieri) [inline features/PJS002_Project-declaration.feature:12]:
```suggestion
Scenario Outline: Validating correct elements relationship from IfcProject
```

**Review by evandroAlfieri — CHANGES_REQUESTED** [review]

File: features/PJS002_Project-declaration.feature (hunk 2 above)
Comment (Ghesselink) [reply features/PJS002_Project-declaration.feature]: https://github.com/buildingSMART/ifc-gherkin-rules/pull/204/commits/6b511ea9a6976d7cf0d1aaa77eb1863fa0d37808

**Review by evandroAlfieri — APPROVED** [review]

**Review by civilx64 — CHANGES_REQUESTED** [review]
All test files need to be renamed to properly reflect the direction of the relationship.  For example it should either be
`project_is_decomposed_by_IfcAlignment` or `IfcAlignment_decomposes_project`.

Otherwise looks good!

**Comment by Ghesselink** [comment]
Yes, this was also the source of another error. I mistakenly thought that the relationship decomposes - isDecomposedBy is reversed from what it actually is. This correction is included in https://github.com/buildingSMART/ifc-gherkin-rules/commit/70b22720a833008d2a1dc2510cf859d72433598f.

**Review by civilx64 — APPROVED** [review]

**Pull request merged**